### ISSUE TITLE
The configuration of Entity Listeners is confusing

### ISSUE THREAD

**Issue by ntoniazzi** [doctrine/DoctrineBundle#732]
The [documentation](http://symfony.com/doc/current/bundles/DoctrineBundle/entity-listeners.html) for configuring Entity Listeners is not clear to me. It states that starting with doctrine/orm 2.5, you don't need to declare the `@ORM\EntityListeners` in the entity.

But then, how is the Listener linked to the Entity (I couldn't make it work without the annotation).

I also tried to configure the listeners in the yaml file (path `doctrine.orm.entity_managers.{manager}.entity_listeners.entities`) without success.

Am I missing something?

Thanks.

**Comment by paulj3** [comment]
@ntoniazzi - agreeed. I am trying to register a listener with a service dependency with no luck.

**Comment by kimhemsoe** [comment]
We always welcome pull requests :)

**Comment by kimhemsoe** [comment]
Without more information there is not much i can do help with. The best i can do is to point you in the direction of the tests, which have some examples.

**Comment by kimhemsoe** [comment]
dup of https://github.com/doctrine/DoctrineBundle/issues/457

**Comment by Steveb-p** [comment]
@kimhemsoe this is probably not a documentation issue. I'm having trouble registering an EntityListener with some dependencies myself and failed to do so. I've went back to using EventSubscriber instead.

I've tried looking into `EntityListenerPass` and checking if it registers to the proper resolver, and sure enough it does. However, whether I specify `@ORM\EntityListeners` annotation with my class inside or not - events were not firing :(
